en/02_Developer_Guides/02_Controllers/02_Routing.md

```diff
@@ -303,6 +303,53 @@ class BreadAPIController extends Controller
 [/alert]
 
+## Nested RequestHandlers
+
+Nested [`RequestHandler`](api:SilverStripe\Control\RequestHandler) routing is used extensively in the CMS and is used to create URL endpoints without yml configuration.  Nesting is done by returning a [`RequestHandler`](api:SilverStripe\Control\RequestHandler) from an action method on another [`RequestHandler`](api:SilverStripe\Control\RequestHandler), usually a [`Controller`](api:SilverStripe\Control\Controller).
+
+`RequestHandler` is the base class for other classes that can handle HTTP requests such as [`Controller`](api:SilverStripe\Control\Controller), [`FormRequestHandler`](api:SilverStripe\Forms\FormRequestHandler) (used by [`Form`](api:SilverStripe\Forms\Form)) and [`FormField`](api:SilverStripe\Forms\FormField).
+
+### How it works
+
+[`Director::handleRequest()`](api:SilverStripe\Control\Director::handleRequest()) begins the url parsing process by parsing the start of the URL and workng out which [`RequestHandler`](api:SilverStripe\Control\RequestHandler) subclass to use, looking in:
+
+- Routes set in yml config under Director `rules`
+- For `/admin` routes specifically, the `$url_segment` config for subclasses of `LeftAndMain`
+
+When a [`RequestHandler`](api:SilverStripe\Control\RequestHandler) matching the first portion of the URL is found, the `handleRequest()` method on the matched [`RequestHandler`](api:SilverStripe\Control\RequestHandler) subclass is called. This passes control to the matched [`RequestHandler`](api:SilverStripe\Control\RequestHandler) and the URL is effectively truncated from the left, removing the portion that lead to ths point.
+
+From there regular request handling occurs and URL will be check to see if it matches `$allowed_actions` on the [`RequestHandler`](api:SilverStripe\Control\RequestHandler), possibly routed via `$url_handlers`. If an `$allowed_action` i.e. method on the [`RequestHandler`](api:SilverStripe\Control\RequestHandler) is matched, if that method returns a [`RequestHandler`](api:SilverStripe\Control\RequestHandler), then control will now be passed to this nested [`RequestHandler`](api:SilverStripe\Control\RequestHandler) and the URL is again effectively truncated from the left.
+
+### Example of a nested RequestHandler being returned in an action method
+
+- There is yml routing directing the path `one` to `RequestHandlerOne`
+- There is a class `RequestHandlerOne` with `$allowed_actions = ['two']`, and a `two()` method returning `RequestHandlerTwo::create();`
+- There is a class `RequestHandlerTwo` with `$allowed_actions = ['hello']`, and a `hello()` method returning `HTTPResponse::create()->setBody('hello');`
+- Navigating to the URL `/one/two/hello` will return a response with a body of "hello"
+
+### Form, HasRequestHandler, FormRequestHandler, and FormField
```

Comment:
```suggestion
### `Form`, `HasRequestHandler`, `FormRequestHandler`, and `FormField`
```
Class names in backticks. Though in this case it's probably better to have a heading that tells you what this section will be about rather than just some class names:

```suggestion
### How `RequestHandler` and `Form` work together
```

Reply:
Updated